packages/headless/src/features/generated-answer/generated-answer-state.ts

```diff
@@ -1,12 +1,37 @@
 import {GeneratedAnswerCitation} from '../../api/generated-answer/generated-answer-event-payload';
 
 export interface GeneratedAnswerState {
+  /**
+   * Determines if the generated answer is loading.
+   */
   isLoading: boolean;
+  /**
+   * Determines if the generated answer is streaming.
+   */
   isStreaming: boolean;
+  /**
+   * The generated answer.
+   */
   answer?: string;
+  /**
+   * The citations used to generate the answer.
+   */
   citations: GeneratedAnswerCitation[];
```

Comment:
Is it expected that the answer can be undefined but not the citations? It'll always be an empty array?

Reply:
correct